wireapp/wire-ios-mono
# feat: Implement Folder Picker UI & Wire-iOS Integration - WPB-12011

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-12011" title="WPB-12011" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-12011</a>  [iOS] Folder Picker View
  </td></table>
  <br />
 







This PR implements the SwiftUI-based folder picker UI and its integration with Wire-iOS, building upon the previously added domain types. 

## Changes

### UI Package (WireMoveToFolderUI)

- Add SwiftUI views for folder picker:
  - `FolderPicker`: Main view with navigation and empty state handling
  - `FolderRow`: Individual folder row with selection state
- Add `FolderPickerHostingController` to bridge SwiftUI with UIKit

### Wire-iOS Integration
- Add adapters to bridge Wire types with UI package:
  - `Conversation+Wire`: Convert ZMConversation to Conversation
  - `WireFolderDirectoryMapper`: Bridge directory operations
- Add `FolderPickerBuilder` to compose and create the UI
- Update folder picker presentation to use new implementation

## UI/UX
- Maintains existing folder picker functionality
- Uses Wire's semantic colors and styling
- Handles empty state and loading
- Proper error handling for folder operations

## To Test
1. Open conversation context menu
2. Select "Move to Folder"
3. Verify existing folders are listed
4. Verify current folder is highlighted
5. Select a different folder
6. Verify conversation moves to selected folder

---

### Checklist


---
